io/include/pcl/io/real_sense_2_grabber.h

```diff
@@ -0,0 +1,213 @@
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef PCL_IO_REAL_SENSE_2_GRABBER_H
+#define PCL_IO_REAL_SENSE_2_GRABBER_H
+
+#include <pcl/io/boost.h>
+#include <pcl/io/grabber.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+
+#include <librealsense2/rs.hpp>
+
+namespace pcl
+{
+
+  /** \brief Grabber for Intel Realsense 2 SDK devices (D400 series)
+    * \note Device width/height defaults to 424/240, the lowest resolutions for D400 devices.
+    * \note Testing on the in_hand_scanner example we found the lower default resolution allowed the app to perform adequately.
+    * \note Developers should use this resolution as a starting point and gradually increase to get the best results
+    * \author Patrick Abadi <patrickabadi@gmail.com>, Daniel Packard <pack3754@gmail.com>
+    * \ingroup io
+  */
+  class PCL_EXPORTS RealSense2Grabber : public pcl::Grabber
+  {
+  public:
+    /** \brief Constructor
+    * \param[in] file_name_or_serial_number used for either loading bag file or specific device by serial number
+    */
+    RealSense2Grabber ( const std::string& file_name_or_serial_number = "", const bool repeat_playback = true );
+
+    /** \brief virtual Destructor inherited from the Grabber interface. It never throws. */
+    virtual ~RealSense2Grabber () throw ();
+
+    /** \brief Set the device options
+    * \param[in] width resolution
+    * \param[in] height resolution
+    * \param[in] fps target frames per second for the device
+    */
+    inline void
+    setDeviceOptions ( uint32_t width, uint32_t height, uint32_t fps = 30 )
+    {
+      device_width_ = width;
+      device_height_ = height;
+      target_fps_ = fps;
+
+      ReInitialize ();
+    }
+
+    /** \brief Start the data acquisition. */
+    virtual void
+    start ();
```

Comment:
We need to decide if this is meant to be `virtual` or simply we want to actually change it to make use of the `override` qualifier.

Reply:
I forgot to add. I believe the use case intended here is `override`. If we don't include include the `virtual` qualifier  and later decide that we need to add it, I believe we'll only be breaking the ABI, which is not considerably harmful given our versioning policy. I also don't expect the to common use case for this class to be people deriving from it and wanting to modify its functionality.